overlord/snapstate/snapmgr.go

```diff
@@ -43,26 +44,33 @@ type SnapManager struct {
 type SnapSetup struct {
 	Name      string `json:"name"`
 	Developer string `json:"developer"`
-	Version   string `json:"version"`
+	Revision  int    `json:"revision"`
 	Channel   string `json:"channel"`
 
-	OldName    string `json:"old-name"`
-	OldVersion string `json:"old-version"`
+	// XXX: strawman is to have this here
+	SideInfo *snap.SideInfo `json:"blessed-metadata"`
```

Comment:
Sounds fine. We'll indeed need data to persist it once we reach the proper installation task, and then we can move it elsewhere onto the state itself.

Just s/blessed-metadata/side-info/ please.